tests/jballerina-unit-test/src/test/resources/test-src/expressions/access/member_access.bal

```diff
@@ -771,6 +771,41 @@ public function testNestedMemberAccessOnIntersectionTypes() {
     assertEquality((), v3);
 }
 
+type myrec record {|
+    string name;
+|};
+
+const LENGTH = 10;
+
+function testMemberAccessWithBinaryExprAsIndex() {
+    int[3] intArr = [0, 1, 2];
+    assertEquality(1, intArr[10 - 9]);
```

Comment:
Instead of passing these in-line shall we assign it to a variable of the relevant type and pass it? So that that bit is also validated.

We can write negative tests with tuples and records also.